lib/utils/proxyAgent.js

```diff
@@ -80,8 +79,12 @@ class HttpsProxyAgent extends Agent {
     const { proxy, secureProxy } = this;
 
     // Wait until the proxy is ready to initialize
-    if (this.initializationPromise instanceof Promise) {
-      await this.initializationPromise;
+    if (
+      this._isCertificateInvalidCallback != null &&
+      this._isCertificateInvalidCallback() &&
+      this._updateCertificateCallback != null
```

Comment:
The safest check here is to use `typeof variable === 'function'`, could be worth adding to `utils/datatypes.js`

```js
function isFunction(value) {
	return typeof value === 'function'
}
```

```suggestion
      isFunction(this._isCertificateInvalidCallback) &&
      this._isCertificateInvalidCallback() &&
      isFunction(this._updateCertificateCallback)
```